### PR TITLE
Implemented logging for socket error handling

### PR DESCRIPTION
**Pull request by ChaseBosman**
Implemented logging to log exceptions to "error_log.txt",  issue #60 . Hopefully this is similar to what you had an idea of, if you have any recommendations or advice regarding my changes please let me know! Thanks.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -3,10 +3,20 @@
...
 
+# Create a logging instance
+logger = logging.getLogger('mainErrorLogging')
+logger.setLevel(logging.INFO)
+# Assign a file-handler to the logging instance and set the level at which it logs
+fh = logging.FileHandler("error_log.txt")
+fh.setLevel(logging.INFO)
+# add file handler to logger
+logger.addHandler(fh)
+
```
Comment (svimanet) [inline main.py:19]: Move this to a new Bot.create_and_return_logger() method

**Comment by svimanet** [comment]
When everything else is handled in the class, I don't think this should be handled on a global scope. Instead, I suggest moving the logger variable into the class constructor as a self variable, and using a method for setting the configuration outside the constructor. This is more consistent with the way things have been done.

I also think that the logger would be more useful if everything is logged with a timestamp. This could be fairly easy with the Time module in py.

Suggestion:
- Add a self.create_and_return_logger() method in Bot class
- Add a self.logger variable in constructor, initiated with the self.create_and_return_logger method
- Add timestamp to logged messages
- Change logger calls to use "self.logger" instead of global "logger"
- Remove global logger declaration

```suggestion
self.logger = self.create_and_return_logger()

...

def create_and_return_logger(self):
    logger = logging.getLogger('mainErrorLogging')
    logger.setLevel(logging.INFO)
    fh= logging.fileHandler("error_log.txt")
    fh.setLevel(logging.INFO)
    logger.addHandleR(fh)
    return logger
´´´

**Comment by ChaseBosman** [comment]
Gotcha, I can see how that definitely works better in terms of a design aspect. It's my first time utilizing exception handling in Python so I was kinda learning as I went... Ill go ahead and submit the changes/reply back here once I've completed this. Thanks!

**Comment by ChaseBosman** [comment]
Hey @svimanet maybe you can help me understand this...when you click on the commit that you merged into the master branch here, it says that I committed it 5 hours ago. When you click on my old pull requests that you have previously merged, you appear there instead of me and it says that you merged it, not me. It's particularly interesting because I noticed that my forked branch still says that it is 9 commits ahead of the master branch...any idea what causes the differences going on?  Edit: Maybe it is that this pull request was merged, but never approved...?

**Comment by svimanet** [comment]
@ChaseBosman sorry for the late response time, I don't always think to swing by here after work. Not entirely sure what you were referring to. When I merge PR, I am marked as responsible for the merge of the request into master, but you are still the committer of the commits being merged in the request. Opening the PR is not the same as merging it. Currently I'm the only one able to merge pull requests in this project. I recently began squashing commits when I merge a PR, maybe this could have caused some confusion?

When it comes to your fork, that is not automatically updated to follow master. So when it says you are 6 commits ahead of master, I believe that means you have committed 6 times since the last master pull. Pulling master should resolve that.

**Comment by ChaseBosman** [comment]
@svimanet Yes, I believe it may be the fact that you squashed the commits upon merge. The interesting thing is that I had not made any commits since I submitted the PR. Like you mentioned, I believe squashing the commits just made it behave a bit different than I'm used to. Thanks!

**Pull request merged**